src/fancy-tools/brushTool.js

```diff
@@ -0,0 +1,295 @@
+import EVENTS from './../events.js';
+import external from './../externalModules.js';
+import baseBrushTool from './../base/baseBrushTool.js';
+import toolColors from './../stateManagement/toolColors.js';
+// Utils
+import getCircle from './shared/brushUtils/getCircle.js';
+import { drawBrushPixels } from './shared/brushUtils/drawBrush.js';
+import isToolActive from '../fancy-tools/shared/isToolActive.js';
+import KeyboardController from './shared/KeyboardController.js';
+// State
+import { getToolState, addToolState } from './../stateManagement/toolState.js';
+import { COLOR_MAP_ID } from '../stateManagement/brushToolColors.js';
+
+const cornerstone = external.cornerstone;
+
+/* Safari and Edge polyfill for createImageBitmap
+ * https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/createImageBitmap
+ */
+
+ if (!('createImageBitmap' in window)) {
+   window.createImageBitmap = async function (imageData) {
+     return new Promise((resolve) => {
+       const img = document.createElement('img');
+
+       img.addEventListener('load', function () {
+
+         resolve(this);
+       });
+
+       const conversionCanvas = document.createElement('canvas');
+
+       conversionCanvas.width = imageData.width;
+       conversionCanvas.height = imageData.height;
+
+       const conversionCanvasContext = conversionCanvas.getContext('2d');
+
+       conversionCanvasContext.putImageData(imageData, 0, 0, 0, 0, conversionCanvas.width, conversionCanvas.height);
+       img.src = conversionCanvas.toDataURL();
+     });
+   };
+ }
+
+export default class extends baseBrushTool {
+  constructor (name = 'brush') {
+    super({
+      name,
+      supportedInteractionTypes: ['mouse'],
+      configuration: defaultBrushToolConfiguration()
```

Comment:
If it's a default, we can have it set in the `baseBrushTool` and use `Object.assign({}, defaultBrushToolConfiguration, customBrushToolConfiguration)` so long as we keep the object flat?

Reply:
Its the default for the brushTool, the brushEraserTool has distinct defaults. Also the keybinds are in the configuration, which i'd rather not flatten, maybe we need to discuss the best approach for this.